src/test/java/io/github/togar2/fluids/test/Main.java

```diff
@@ -42,7 +42,7 @@ public static void main(String[] args) {
 					handler.placeFluid(instance, event.getBlockPosition(), MinestomFluids.WATER.getDefaultState());
 				} else {
 					event.getInstance().placeBlock(new BlockHandler.Placement(
-							Block.WATER, event.getInstance(), event.getBlockPosition().relative(event.getBlockFace())));
+							Block.WATER, event.getBlock(), event.getInstance(), event.getBlockPosition().relative(event.getBlockFace())));
```

Comment:
I am pretty sure event.getBlock() does not return the right block. You should take the current block at the placement position (this returns the block at the interaction position)